spec/services/shopify_refunder_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'spec_helper'
+
+require_relative '../../lib/active_merchant/billing/gateways/shopify.rb'
+
+describe ShopifyRefunder do
+  let(:transaction_id) { '0xDEADBEEF' }
+  let(:options) { { reason: 'Actual reason', order_id: transaction_id } }
+  let(:transaction_amount) { 1 }
+  let(:credited_money_in_cents) { 100 }
+  let(:transaction) { double(:transaction, amount: transaction_amount, id: transaction_id) }
+
+  before do
+    allow(ShopifyAPI::Transaction).to receive(:find).and_return(transaction)
+  end
+
+  context '.initialize' do
+    subject { described_class.new(credited_money_in_cents, transaction_id, options) }
```

Comment:
What if one of the required parameters is not entered, we raise an error?

For example, the credited_money_in_cents is a required parameter in the method above. If the user is allowed to enter whatever params they want, then we need to check the required parameters and raise an error if they are not filled? I know how to achieve that, but is there an example of something you have implemented that follows that behavior?